server/rest/src/main/java/org/infinispan/rest/resources/ServerResource.java

```diff
@@ -466,6 +473,30 @@ private CompletionStage<RestResponse> dataSourceTest(RestRequest restRequest) {
       }, invocationHelper.getExecutor());
    }
 
+   private CompletionStage<RestResponse> getCacheConfigDefaultAttributes(RestRequest request) {
+      Configuration configuration = new ConfigurationBuilder().build();
+      Map<String, String> attributes = new LinkedHashMap<>();
+      allAttributes(configuration, attributes);
+      return asJsonResponseFuture(Json.make(attributes), isPretty(request));
+   }
+
+   private static void allAttributes(ConfigurationElement<?> element, Map<String, String> attributes) {
+      for (Attribute<?> attribute : element.attributes().attributes()) {
+         AttributeDefinition<?> definition = attribute.getAttributeDefinition();
+         String value;
+         if(attribute.getInitialValue() == null){
+            value = null;
+         } else {
+            value = attribute.getInitialValue().toString();
+         }
+         attributes.put(definition.name(), value);
+      }
+      for (ConfigurationElement<?> child : element.children()) {
+         allAttributes(child, attributes);
```

Comment:
I just noticed this is recursively calling itself, however we don't have a prefix or anything. Are we sure no property is being overridden by a sub element? For example if we have a "timeout" setting that is across multiple attribute definitions only the last will show and not know which it applies to.

Reply:
I had forgotten about this case, but I have now updated the PR to reflect the necessary changes.

Now the API will output a JSON like: 
```
{
  "clustering": {
    "mode": "LOCAL",
    "remote-timeout": "15000",
    ...
  },
  "hash": {
    "consistent-hash-factory": null,
    "owners": "2",
   ...
  },
  ...
}
```